### PR TITLE
Fix for no user answers in TestAnswers

### LINKED ISSUE #467: [DEV] After entering 'Statystyki' page, error is visible
1. Login as a user
2. Do not answer for any available question
3. Go to the page : https://dev.devadventcalendar.pl/Manage/DisplayStatistics

Actual Results:

An unhandled exception occurred while processing the request.
InvalidOperationException: Sequence contains no elements.
lambda_method(Closure , QueryContext , DbDataReader , ResultContext , int[] , ResultCoordinator )

System.InvalidOperationException: Sequence contains no elements.
   at lambda_method(Closure , QueryContext , DbDataReader , ResultContext , Int32[] , ResultCoordinator )
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression)
   at System.Linq.Queryable.Max[TSource,TResult](IQueryable`1 source, Expression`1 selector)
   at DevAdventCalendarCompetition.Repository.StatisticsRepository.GetAnsweredCorrectMaxTestId(String userId) in /app/DevAdventCalendarCompetition.Repository/StatisticsRepository.cs:line 44
   at DevAdventCalendarCompetition.Services.StatisticsService.GetHighestTestNumber(String userId) in /app/DevAdventCalendarCompetition.Services/StatisticsService.cs:line 43
   at DevAdventCalendarCompetition.Services.StatisticsService.FillResultsWithTestStats(String userId) in /app/DevAdventCalendarCompetition.Services/StatisticsService.cs:line 22
   at DevAdventCalendarCompetition.Controllers.ManageController.DisplayStatistics() in /app/DevAdventCalendarCompetition/Controllers/ManageController.cs:line 368
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.HttpOverrides.HttpMethodOverrideMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.MigrationsEndPointMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

### PR DESCRIPTION
**Pull request by BlackMorzan**
## Motivation and context

fixes #467

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Repository/StatisticsRepository.cs [hunk 1]
```diff
@@ -27,11 +27,6 @@ public int GetUserTestWrongAnswerCount(string userId, int testId)
 
         public DateTime? GetUserTestCorrectAnswerDate(string userId, int testId)
         {
-            if (!this._dbContext.Tests.Any())
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Repository/StatisticsRepository.cs:30]: it was ok

**Comment by WTobor** [comment]
when there are found issues like this, there is a good approach to create a test to check the logic

**Comment by BlackMorzan** [comment]
> when there are found issues like this, there is a good approach to create a test to check the logic

I agree, but since functions that we want to test works directly on DB it would require to create integration tests and I don't believe I can make them till tomorrow.

**Comment by WTobor** [comment]
@BlackMorzan ok, so please put it in new PR

**Review by WTobor — APPROVED** [review]

**Pull request merged**